src/nvim/eval.c

```diff
@@ -21628,25 +21624,28 @@ static void func_do_profile(ufunc_T *fp)
 {
   int len = fp->uf_lines.ga_len;
 
+  if (!fp->uf_prof_initialized) {
   if (len == 0)
     len = 1;      /* avoid getting error for allocating zero bytes */
-  fp->uf_tm_count = 0;
-  fp->uf_tm_self = profile_zero();
-  fp->uf_tm_total = profile_zero();
+    fp->uf_tm_count = 0;
+    fp->uf_tm_self = profile_zero();
+    fp->uf_tm_total = profile_zero();
```

Comment:
indent does not match line 21628

Reply:
Thanks, it was actually completely broken.